### DISCUSSION TITLE
Standardization of element annotations

### DISCUSSION THREAD

**Discussion by panars** [DIGGSml/schema-dev#25] [Q&A] [answered]
How are the values of elements standardized? Or are they, maybe not.
For example, the DrivenPenetrationTest is described as _An object describing a driven penetration test procedure. The Driven Penetration Test includes all methods that involve driving a rod (with or without a sampler) by impact hammer. The most common test of this type inclue the Standard Penetration Test and the Becker Hammer Test._
What is to keep one record for a standard penetration test using a value of SPT and another record using a value of Standard Penetration Test?
<diggs_geo:penetrationTestType>SPT</diggs_geo:penetrationTestType> 
<diggs_geo:penetrationTestType>Standard Penetration Test</diggs_geo:penetrationTestType>

**Comment by dponti** [comment, marked as answer]
Hi Peter - There is no specific control within the schema to restrict a specific set of terms to use for the penetrationTestType property. This value has a type of gml:CodeType, which means that the value is intended to come from a controlled list of terms, whereby the meaning can be looked up from an online dictionary or where some authority has defined the term so that, in that context, the term is universally understood. The controlled list or authority is contained within the optional codeSpace attribute of the element. So for example, if the term SPT is defined in a dictionary at http://diggsml.org/terms/penetrationTestTypes.xml, the usage would be:

    <diggs_geo:penetrationTestType codeSpace = "http://diggsml.org/terms/penetrationTestTypes.xml#SPT">Standard Penetration Test</diggs_geo:penetrationTestType>

where #SPT points to the id of the dictionary entry that defines Standard Penetration Test.

There are many properties in DIGGS where best practice would be to use terms defined from a controlled (external) list. This allows for the users to use familiar terms and provide a definition for others to have.. With the exception of a few cases (test result parameters for example), the schema does not require use of the codespace attribute. Where it is important to use strict terminology, we can change the type of this property to gml:CodeWithAuthorityType (which requires use of the codeSpace attribute), or as a type with defined enumerations (an enumerated list), which then would require schema changes if the terms are altered in the future.

For the present time, we've defined things in this way in DIGGS with the idea that as we review and use the schema more widely, we can begin to develop appropriate codeList dictionaries and/or decide to place more restrictions on these types of properties via enumerated lists.
